bundle/edu.gemini.pot/src/main/scala/edu/gemini/spModel/target/SPTarget.scala

```diff
@@ -0,0 +1,205 @@
+
+  def setName(s: String): Unit =
+    setNewTarget(Target.name.set(target, s))
+
+  def setRaDegrees(value: Double): Unit =
+    ra.set(target, RightAscension.fromDegrees(value)).foreach(setNewTarget)
+
+  def setRaHours(value: Double): Unit =
+    ra.set(target, RightAscension.fromHours(value)).foreach(setNewTarget)
+
+  def setRaString(hms: String): Unit =
+    for {
+      a <- Angle.parseHMS(hms).toOption
+      t <- ra.set(target, RightAscension.fromAngle(a))
+    } setNewTarget(t)
+
+  def setDecDegrees(value: Double): Unit =
+    for {
+      d <- Declination.fromDegrees(value)
+      t <- dec.set(target, d)
+    } setNewTarget(t)
+
+  def setDecString(dms: String): Unit =
+    for {
+      a <- Angle.parseDMS(dms).toOption
+      d <- Declination.fromAngle(a)
+      t <- dec.set(target, d)
+    } setNewTarget(t)
+
+  def setRaDecDegrees(ra: Double, dec: Double): Unit =
+    for {
+      cs <- Coordinates.fromDegrees(ra, dec)
+      t  <- Target.coords.set(target, cs)
+    } setNewTarget(t)
+
+  def setSpectralDistribution(sd: Option[SpectralDistribution]): Unit =
+    Target.spectralDistribution.set(target, sd).foreach(setNewTarget)
+
+  def setSpatialProfile(sp: Option[SpatialProfile]): Unit =
+    Target.spatialProfile.set(target, sp).foreach(setNewTarget)
+
+  def getSpectralDistribution: Option[SpectralDistribution] =
+    Target.spectralDistribution.get(target).flatten
+
+  def getSpatialProfile: Option[SpatialProfile] =
+    Target.spatialProfile.get(target).flatten
+
+  def isTooTarget: Boolean =
+    target.fold(_ => true,  _ => false, _ => false)
+
+  def isSidereal: Boolean =
+    target.fold(_ => false, _ => true,  _ => false)
+
+  def isNonSidereal: Boolean =
+    target.fold(_ => false, _ => false, _ => true)
+
+  def getCoordinates(when: Option[Long]): Option[Coordinates] =
+    target.fold(
+      _ => None,
+      s => Some(s.coordinates),
+      n => when.flatMap(n.coords)
+    )
+
+  // Accessors for Java that use Gemini's Option type and boxed primitives
+
+  type GOLong   = GOption[java.lang.Long]
+  type GODouble = GOption[java.lang.Double]
+
+  private def gcoords[A](when: GOLong)(f: Coordinates => A): GOption[A] =
+    getCoordinates(when.asScalaOpt.map(_.longValue())).map(f).asGeminiOpt
+
+  def getRaHours(time: GOLong): GODouble =
+    gcoords(time)(_.ra.toHours)
+
+  def getRaDegrees(time: GOLong):GODouble =
+    gcoords(time)(_.ra.toDegrees)
+
+  def getRaString(time: GOLong): GOption[String] =
+    gcoords(time)(_.ra.toAngle.formatHMS)
+
+  def getDecDegrees(time: GOLong): GODouble =
+    gcoords(time)(_.dec.toDegrees)
+
+  def getDecString(time: GOLong): GOption[String] =
+    gcoords(time)(_.dec.toAngle.formatDMS)
+
+  def getSkycalcCoordinates(time: GOLong): GOption[SCoordinates] =
+    gcoords(time)(cs => new SCoordinates(cs.ra.toDegrees, cs.dec.toDegrees))
+
+  def getSiderealTarget: Option[SiderealTarget] =
+    target.fold(_ => None, Some(_), _ => None)
+
+  def putNewMagnitude(mag: Magnitude): Unit =
```

Comment:
They will be renamed now; for a while there were methods using  both new and old magnitudes.